sdk/search/Azure.Search.Documents/tests/Models/SearchOptionsTests.cs

```diff
@@ -253,10 +253,10 @@ public void QueryCaptionOption()
 
             // We can also set `QueryCaption` to a value unknown to the SDK.
             searchOptions.QueryCaption = "unknown";
-            Assert.AreEqual($"unknown|highlight-True", searchOptions.QueryCaptionRaw);
```

Comment:
Do we want this to happen? Or should we modify the code to allow for passing of `highlight` parameter when `unknown` is passed? I couldn't test whether the SDK is fine with accepting an `unknown|highlight-True` payload.

@ShivangiReja thoughts?

Reply:
No, we want to allow the `highlight` parameter to be passed only when the `QueryCaptionType` is set to `Extractive`. The code is correct in this PR.